marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/keystore/AbstractKeyStoreInit.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2012-2020 MarkLogic Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package com.marklogic.hub.central.keystore;
+
+import org.springframework.boot.autoconfigure.web.ServerProperties;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.server.Ssl;
+import org.springframework.boot.web.server.SslStoreProvider;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+
+import java.io.InputStream;
+import java.security.KeyStore;
+
+@Profile({"azure", "aws"})
+public abstract class AbstractKeyStoreInit implements KeyStoreInit {
+
+    public static String KEY_STORE_TYPE = "PKCS12";
+
+    @Bean
+    public ServerProperties serverProperties() {
+        final ServerProperties serverProperties = new ServerProperties();
+        final Ssl ssl = new Ssl();
+        final String keystorePassword = getKeyStorePassword();
+        System.setProperty("server.ssl.key-store-password", keystorePassword);
+        ssl.setKeyPassword(keystorePassword);
+        serverProperties.setSsl(ssl);
+        return serverProperties;
+    }
+
+    @Bean
```

Comment:
"@Bean" is not inheritable, so I'm curious how these would work.

I think the better approach is to declare both of these Bean methods on AWSConfiguration and AzureConfiguration, and then move this reusable logic into a CloudProviderUtil class in the com.marklogic.hub.central.cloud package (with the AWS class in cloud.aws and the Azure class in cloud.azure). 

Generally, for reusing functionality, it's better to use composition instead of inheritance. Inheritance can be handy for reusing state - i.e. fields defined in a parent class. 

CloudProviderUtil would then have two methods - buildServerProperties(keyStorePassword) and configureSslStoreProvider(tomcatServletWebServerFactory, keyStorePassword, keyStoreFile).